graphql_compiler/compiler/ir_lowering_match/ir_lowering.py

```diff
@@ -93,44 +93,98 @@ def visitor_fn(expression):
+            # We transform any structures that resemble the following:
+            #    BinaryComposition(u'has_substring', X, Y)
+            # into the following:
+            #    BinaryComposition(
+            #        u'LIKE',
+            #        X,
+            #        BinaryComposition(
+            #            u'+',
+            #            Literal("%"),
+            #            BinaryComposition(
+            #                 u'+',
+            #                 Y,
+            #                 Literal("%")
+            #            )
+            #        )
+            #    )
+            return BinaryComposition(
+                u'LIKE',
+                expression.left,
+                BinaryComposition(
+                    u'+',
+                    Literal('%'),
+                    BinaryComposition(
+                        u'+',
+                        expression.right,
+                        Literal('%')
+                    )
+                )
+            )
+        elif expression.operator == u'starts_with':
+            # Rewrite BinaryComposition expressions with "starts_with" into representable form.
+            # The implementation of "starts_with" must use the LIKE operator in MATCH, and must
+            # append "%" symbols to the substring being matched.
+            # We transform any structures that resemble the following:
+            #    BinaryComposition(u'starts_with', X, Y)
+            # into the following:
+            #    BinaryComposition(
+            #        u'LIKE',
+            #        X,
+            #        BinaryComposition(
+            #           u'+',
+            #           Y,
+            #           Literal("%")
+            #        )
+            #    )
+            return BinaryComposition(
+                u'LIKE',
+                expression.left,
                 BinaryComposition(
                     u'+',
                     expression.right,
                     Literal('%')
                 )
             )
-        )
+        elif expression.operator == u'ends_with':
+            # Rewrite BinaryComposition expressions with "ends_with" into representable form.
+            # The implementation of "ends_with" must use the LIKE operator in MATCH, and must
+            # prepend "%" symbols to the substring being matched.
+            # We transform any structures that resemble the following:
+            #    BinaryComposition(u'ends_with', X, Y)
+            # into the following:
+            #    BinaryComposition(
+            #        u'LIKE',
+            #        X,
+            #        BinaryComposition(
+            #           u'+',
+            #           Literal("%"),
+            #           Y
+            #        )
+            #    )
+            return BinaryComposition(
+                u'LIKE',
+                expression.left,
+                BinaryComposition(
```

Comment:
There's a lot of repetition here, this is literally just the three functions copy-pasted into one function. I was hoping for a refactor that gets rid of all the repetitions. Consider how left-addition of `%` is repeated twice, right-addition of `%` is repeated twice, and putting all of that in a `LIKE` clause is repeated three times – each time also requiring a lengthy comment describing the transformation.

We should be able to define and document some helper functions that compose nicely together and reduce the complexity and repetitiveness of the code.